src/auth/authContext.jsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from "react";
+import firestore, { auth } from '../firebaseConfig';
+
+export const AuthContext = React.createContext({
+  user: {},
+  setUser: () => { },
+});
+
+const AuthProvider = (props) => {
+  const [user, setUser] = useState({});
+
+  useEffect(() => {
+    auth.onAuthStateChanged((auth) => {
+      console.log('AUTHSTATECHANGE ', auth);
+      processAuth(auth,setUser);
+    });
+    return () => {
+      // cleanup
+    }
+  }, []);
+
+  return (
+    <AuthContext.Provider value={{ user, setUser }}>
+      {props.children}
+    </AuthContext.Provider>
+  );
+};
+
+export const processAuth = async (auth,setUser) => {
+
+  if (!auth){
+    return setUser({});
+  }
+
+  let firestoreResult = await firestore
+    .doc("users/" + auth.uid)
```

Comment:
If it is possible can you write small note about why you are using "uid".

Reply:
PTAL 👍